x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/type/EsqlDataTypeConverter.java

```diff
@@ -148,10 +164,75 @@ public static TemporalAmount parseTemporalAmout(Number value, String qualifier,
         };
     }
 
+    /**
+     * Converts a string or date expression to long.
+     */
+    private static Long foldExpressionToLong(Expression exp) {
```

Comment:
The converter should only contain methods that perform one type conversion to another - move the folding in a separate method.
The string to -> conversion will be moved into the planner.

Reply:
I also feel this does not belong to here, will move the folding back to the function.